components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/ClaimProviderImpl.java

```diff
@@ -0,0 +1,123 @@
+import org.wso2.carbon.identity.oidc.session.util.OIDCSessionManagementUtil;
+import org.wso2.carbon.identity.openidconnect.ClaimProvider;
+
+import javax.servlet.http.Cookie;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ * This class is used to insert sid claim into ID token
+ */
+public class ClaimProviderImpl implements ClaimProvider {
+
+    private static Log log = LogFactory.getLog(ClaimProviderImpl.class);
+    private String claimValue;
+    private String name;
+
+
+    @Override
+    public Map<String, Object> getAdditionalClaims(OAuthAuthzReqMessageContext oAuthAuthzReqMessageContext,
+                                                   OAuth2AuthorizeRespDTO oAuth2AuthorizeRespDTO)
+            throws IdentityOAuth2Exception {
+        Map<String, Object> additionalClaims = new HashMap<>();
+        this.name = "sid";
+        OIDCSessionState previousSession = getSessionState(oAuthAuthzReqMessageContext);
+        if (previousSession == null) {
+            // If there is no previous browser session, generate new sid value.
+            claimValue = UUID.randomUUID().toString();
+            if (log.isDebugEnabled()) {
+                log.debug("sid claim found for Back channel Implicit flow ");
+            }
+        } else {
+            // Previous browser session exists, get sid claim from OIDCSessionState.
+            claimValue = previousSession.getSidClaim();
+            if (log.isDebugEnabled()) {
+                log.debug("sid claim is found in the session");
+            }
+        }
+        additionalClaims.put(name, claimValue);
+        return additionalClaims;
+    }
+
+    @Override
+    public Map<String, Object> getAdditionalClaims(OAuthTokenReqMessageContext oAuthTokenReqMessageContext,
+                                                   OAuth2AccessTokenRespDTO oAuth2AccessTokenRespDTO)
+            throws IdentityOAuth2Exception {
+        Map<String, Object> additionalClaims = new HashMap<>();
+        String accessCode = oAuthTokenReqMessageContext.getOauth2AccessTokenReqDTO().getAuthorizationCode();
+        OIDCBackChannelAuthCodeCacheEntry cacheEntry = getOIDCBackChannelAuthCodeCacheEntry(accessCode);
+        if (cacheEntry != null) {
+            claimValue = cacheEntry.getSessionId();
+        }
+        if (claimValue != null) {
+            if (log.isDebugEnabled()) {
+                log.debug("sid claim is found in the session");
+            }
+            additionalClaims.put("sid", claimValue);
+        }
+        return additionalClaims;
+    }
+
+    /**
+     * Return previousSessionState using opbs cookie.
+     *
+     * @param oAuthAuthzReqMessageContext
+     * @return OIDCSession state
+     */
+    private OIDCSessionState getSessionState(OAuthAuthzReqMessageContext oAuthAuthzReqMessageContext) {
+        Cookie[] cookies = oAuthAuthzReqMessageContext.getAuthorizationReqDTO().getCookie();
```

Comment:
improve debug logs in getSessionState().

Need information whether cookie header is not available, or obps cookie not available or session state against the obps cookie is not available